### PR TITLE
Previous scene navigation

### LINKED ISSUE #434: [FEATURE] Back button should take the user to the correct previous scene
### Motivation

- Remove the Battle/LevelManager.cs script, as it's not being used.
- Implement a method in Shared/LevelManager.cs (maybe rename this script?) to make the back button take you back to the scene from which you come from. This will be useful in the case where a scene can be accessed from more than one scene, and you want to go back to the place you came from.

### PR DESCRIPTION
**Pull request by tkz00**
Closes #434 

## Motivation

Currently, the back button relies on a pre-defined scene name to transition, which becomes problematic when the same scene (like the campaignsMap) can be reached from multiple places (Overworld and Dungeon in this case).  For the upcoming dungeon feature, the back button should return the user to the scene they came from, not a fixed scene.  A hardcoded value won't work in this scenario.

## Summary of changes

- New logic for scene navigation has been added, now a stack of the visited scenes is maintained through the lifetime of the application, when a back button is pressed, the scene with the name on top of this stack is loaded, and the name removed from the stack.
- `LevelManager` script has been renamed to `SceneNavigator`, which is a much better fitting name, even more now since it holds the logic of changing scenes, including the back buttons functionality.

## How has this been tested?

Navigate through all the scenes and test the back button on each one is taking you to the previous scene. (Be aware [the Inventory screen is broken](https://github.com/lambdaclass/champions_of_mirra/issues/446))

### CODE REVIEW

File: client/Assets/Scripts/CampaignsMap/CampaignItem.cs [hunk 1]
```diff
@@ -17,15 +17,17 @@ public class CampaignItem : MonoBehaviour
...
 
     // Load campaign scene if its unlocked. Scene needs to have the same name as our campaign object.
     public void SelectCampaign()
     {
+		Debug.Log("selected campaign");
         if (campaignData.status == LevelProgress.Status.Unlocked) {
             CampaignManager.selectedCampaignData = campaignData;
             CampaignManager.campaignPrefab = campaignToShowPrefab;
-            sceneManager.ChangeToSceneWithDelay("Campaign");
+            // sceneNavigator.ChangeToSceneWithDelay("Campaign");
```
Comment (ncontinanza) [inline client/Assets/Scripts/CampaignsMap/CampaignItem.cs:29]:
```suggestion
```

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by ncontinanza — APPROVED** [review]

**Review by ncontinanza — APPROVED** [review]

File: client/Assets/Scripts/Shared/SceneNavigator.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+using System.Collections;
```
Comment (lotuuu) [inline client/Assets/Scripts/Shared/SceneNavigator.cs:1]: This file is not formatted! Saving any changes here formats the file entirely for me (with the setting    `"editor.formatOnSave": true` in vscode). Does it format it differently for you or did you have this setting off?

**Review by lotuuu — CHANGES_REQUESTED** [review]

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Review by lotuuu — APPROVED** [review]
Approving since format issues will be ignored until #456

**Pull request merged**